nsxt/resource_nsxt_policy_domain.go

```diff
@@ -0,0 +1,336 @@
+/* Copyright © 2020 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	gm_infra "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/global_infra"
+	gm_domain "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/global_infra/domains"
+	gm_model "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+// This resource is supported only for Policy Global Manager
+func resourceNsxtPolicyDomain() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyDomainCreate,
+		Read:   resourceNsxtPolicyDomainRead,
+		Update: resourceNsxtPolicyDomainUpdate,
+		Delete: resourceNsxtPolicyDomainDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"location": {
```

Comment:
I think historically we use plural when attribute is specified as a list, not as a clause, so suggest to use "locations" here. Maybe these should be "sites", since these correspond to nsxt_policy_site, is that correct?

Reply:
Right. In the UI is locations, but everywhere else its sites. I'll change it.